### PR TITLE
Fix created/updated post detection to use post_date_gmt

### LINKED ISSUE #11: Dashboard incorrectly classifies posts as created vs updated
## Problem

The dashboard's created/updated post classification compares revision IDs to determine if a post existed before a migration. This is unreliable — a post updated for the first time by a migration won't have a prior revision, causing it to be misclassified as "created."

This affects both the migration detail stats and individual post status in two places:
- `get_post_statuses()` — bulk query for migration post list
- `get_post_migration_data()` — single-post detail for diff/rollback

## Expected behavior

A post should be classified as "updated" if it existed before the migration started, regardless of its revision history.

## Fix

Compare the post's `post_date_gmt` against the migration timestamp:
- `post_date_gmt < migration_timestamp` → "updated"
- Otherwise → "created"

### PR DESCRIPTION
**Pull request by jorostoyanov**
## Summary
- Replace revision-ID-based created/updated classification with `post_date_gmt` comparison against migration timestamp
- Affects both `get_post_statuses()` (bulk query) and `get_post_migration_data()` (single-post detail)
- Fixes edge case where pre-existing posts with no prior revisions were misclassified as "created"

## Test plan
- [x] `test_pre_existing_post_without_prior_revision_is_updated` — post created in 2020, first touched by migration, correctly classified as "updated"
- [x] `test_post_created_during_migration_is_created` — post created during migration, correctly classified as "created"
- [x] Existing `test_get_migration_detail_correct_stats_data` continues to pass

Fixes #11

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: includes/class-nre-migration-dashboard.php [hunk 1]
```diff
@@ -547,11 +546,12 @@ private function get_post_statuses( $term_id, $migration_name, $migration_ts ) {
...
 		);
 
-		$statuses = [];
+		$migration_date = gmdate( 'Y-m-d H:i:s', $migration_ts );
+		$statuses       = [];
 
 		foreach ( $rows as $row ) {
 			$post_id = (int) $row->post_id;
-			$status  = ( (int) $row->min_rev_id < (int) $row->first_migration_rev_id ) ? 'updated' : 'created';
+			$status  = ( $row->post_created_gmt < $migration_date ) ? 'updated' : 'created';
```
[the remark this answers is not in the document]
Comment (jorostoyanov) [reply includes/class-nre-migration-dashboard.php]: Fixed in cc1547b — added zero-date guard to the bulk path, matching the single-post logic.
